frontend/src/components/filters/Filters.js

```diff
@@ -2,604 +2,131 @@ import counterpart from 'counterpart';
 import PropTypes from 'prop-types';
 import React, { PureComponent } from 'react';
 import { connect } from 'react-redux';
-// -- using iMap from immutable
-import { Map as iMap } from 'immutable';
-import _ from 'lodash';
-
-import { DATE_FIELDS } from '../../constants/Constants';
 import {
-  generateMomentObj,
-  getFormatForDateField,
-} from '../widget/RawWidgetHelpers';
-import { parseDateWithCurrentTimezone } from '../../utils/documentListHelper';
-import { fieldValueToString } from '../../utils/tableHelpers';
+  updateWidgetShown,
+  setNewFiltersActive,
+  updateActiveFilter,
+  clearAllFilters,
+  annotateFilters,
+  isFilterValid,
+  updateNotValidFields,
+  parseToPatch,
+} from '../../actions/FiltersActions';
+
 import FiltersFrequent from './FiltersFrequent';
 import FiltersNotFrequent from './FiltersNotFrequent';
-
-/**
- * @method parseDateToReadable
- * @summary ToDo: Describe the method
- * @param {*} widgetType
- * @param {*} value
- * @todo Write the documentation
- */
-export function parseDateToReadable(widgetType, value) {
-  if (DATE_FIELDS.indexOf(widgetType) > -1) {
-    return parseDateWithCurrentTimezone(value, widgetType);
-  }
-  return value;
-}
+import deepUnfreeze from 'deep-unfreeze';
+import { getEntityRelatedId } from '../../reducers/filters';
 
 /**
  * @file Class based component.
  * @module Filters
- * @extends Component
+ * @extends PureComponent
  */
 class Filters extends PureComponent {
-  state = {
-    activeFilter: null,
-    activeFiltersCaptions: null,
-    flatFiltersMap: null,
-    notValidFields: null,
-    widgetShown: false,
-  };
-
-  /**
-   * @method UNSAFE_componentWillReceiveProps
-   * @summary ToDo: Describe the method
-   */
-  UNSAFE_componentWillReceiveProps() {
-    this.parseActiveFilters();
-  }
-
-  /**
-   * @method componentDidMount
-   * @summary ToDo: Describe the method
-   */
-  componentDidMount() {
-    this.parseActiveFilters();
-  }
-
-  /**
-   * @method arrangeFilters
-   * @summary combines the filters and also removes the other actives ones from same group
-   */
-  arrangeFilters = (filterData) => {
-    let combinedFilters = [];
-    for (const [key] of filterData.entries()) {
-      let item = filterData.get(key);
-      if (typeof item.includedFilters !== 'undefined') {
-        combinedFilters.push(...item.includedFilters);
-      } else {
-        combinedFilters.push(item);
-      }
-    }
-    let mappedFiltersData = new Map();
-    combinedFilters.forEach((item) => {
-      mappedFiltersData.set(item.filterId, item);
-    });
-    return mappedFiltersData;
-  };
-
-  // PARSING FILTERS ---------------------------------------------------------
-
-  /*
-   * parseActiveFilters - this function does three things:
```

Comment:
As far as I understand this was divided into smaller functions. Comments were also carried over, right ?